Refactor various parts of caret animation code post-upstreaming

rr-codes opened the pull request:
#### 1041ca48066bb678e125595f2a4aaff90ff65ad3
<pre>
Refactor various parts of caret animation code post-upstreaming
<a href="https://bugs.webkit.org/show_bug.cgi?id=258386">https://bugs.webkit.org/show_bug.cgi?id=258386</a>
rdar://110802729

Reviewed by Mike Wyrzykowski.

Refactors some logic in and around caret animation logic:

* Fixes typo of `dictationCaretAnimatorUpdateRate` inside `DictationCaretAnimator.cpp`
* Simplifies all the various caret color computing logic into `computeCaretColor`
* Renames `CaretAnimatorType::Alternate` to `CaretAnimatorType::Dictation`
* Renames `updateCaretDecorationPlacement` to `updateCursorAccessoryPlacement`
* Removes `currentTimeSinceEpoch` and instead directly uses `MonotonicTime` for the timekeeping variables
* Removes unnecessary `ReducedResolutionSeconds` parameter for various caret animator methods

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::serviceCaretAnimation):
* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::createCaretAnimator):
(WebCore::CaretBase::computeCaretColor):
(WebCore::CaretBase::paintCaret const):
(WebCore::FrameSelection::updateAppearance):
* Source/WebCore/editing/FrameSelection.h:
* Source/WebCore/platform/CaretAnimator.cpp:
(WebCore::CaretAnimator::serviceCaretAnimation):
(WebCore::CaretAnimator::paint const):
* Source/WebCore/platform/CaretAnimator.h:
(WebCore::CaretAnimator::didStart):
(WebCore::currentTimeSinceEpoch): Deleted.
(WebCore::platformCaretColor): Deleted.
* Source/WebCore/platform/DictationCaretAnimator.cpp:
(WebCore::keyframe):
(WebCore::DictationCaretAnimator::keyframeCount const):
(WebCore::DictationCaretAnimator::keyframeTimeDelta const):
(WebCore::DictationCaretAnimator::updateAnimationProperties):
(WebCore::DictationCaretAnimator::start):
(WebCore::DictationCaretAnimator::paint const):
* Source/WebCore/platform/DictationCaretAnimator.h:
* Source/WebCore/platform/OpacityCaretAnimator.cpp:
(WebCore::OpacityCaretAnimator::updateAnimationProperties):
(WebCore::OpacityCaretAnimator::start):
(WebCore::OpacityCaretAnimator::paint const):
* Source/WebCore/platform/OpacityCaretAnimator.h:
* Source/WebCore/platform/SimpleCaretAnimator.cpp:
(WebCore::SimpleCaretAnimator::updateAnimationProperties):
(WebCore::SimpleCaretAnimator::start):
* Source/WebCore/platform/SimpleCaretAnimator.h:
* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::autocorrectionReplacementMarkerColor const):
* Source/WebKit/Platform/cocoa/_WKWebViewTextInputNotifications.mm:
(-[_WKWebViewTextInputNotifications dictationDidStart]):
(-[_WKWebViewTextInputNotifications dictationDidResume]):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::selectionDidChange):
(WebKit::WebViewImpl::updateCursorAccessoryPlacement):
(WebKit::WebViewImpl::updateCaretDecorationPlacement): Deleted.
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::getPlatformEditorState const):
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame caretColor]):

Canonical link: <a href="https://commits.webkit.org/265430@main">https://commits.webkit.org/265430@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fb1e6d04a1fb0060824b3cc81f1a9b3540a7de57

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10783 "Passed style check") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11003 "Failed to compile WebKit") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11304 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12432 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/10341 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10798 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13377 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10980 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13239 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10943 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11849 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/9069 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12836 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9143 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9726 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16980 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10214 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9878 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13125 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10354 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/8441 "14 flakes 4 failures") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9512 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2607 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13785 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10215 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/editing/FrameSelection.cpp:
```diff
@@ -1898,15 +1898,35 @@ void FrameSelection::paintCaret(GraphicsContext& context, const LayoutPoint& pai
     if (elementStyle.hasAutoCaretColor())
         return { };
     return elementStyle.colorResolvingCurrentColor(elementStyle.caretColor());
+#elif HAVE(REDESIGNED_TEXT_CURSOR)
```
For some reason which codepath we take on which platform was a bit hard to reason about 😅

rr-codes committed:
Refactor various parts of caret animation code post-upstreaming
https://bugs.webkit.org/show_bug.cgi?id=258386
rdar://110802729

Reviewed by Mike Wyrzykowski.

Refactors some logic in and around caret animation logic:

* Fixes typo of `dictationCaretAnimatorUpdateRate` inside `DictationCaretAnimator.cpp`
* Simplifies all the various caret color computing logic into `computeCaretColor`
* Renames `CaretAnimatorType::Alternate` to `CaretAnimatorType::Dictation`
* Renames `updateCaretDecorationPlacement` to `updateCursorAccessoryPlacement`
* Removes `currentTimeSinceEpoch` and instead directly uses `MonotonicTime` for the timekeeping variables
* Removes unnecessary `ReducedResolutionSeconds` parameter for various caret animator methods

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::serviceCaretAnimation):
* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::createCaretAnimator):
(WebCore::CaretBase::computeCaretColor):
(WebCore::CaretBase::paintCaret const):
(WebCore::FrameSelection::updateAppearance):
* Source/WebCore/editing/FrameSelection.h:
* Source/WebCore/platform/CaretAnimator.cpp:
(WebCore::CaretAnimator::serviceCaretAnimation):
(WebCore::CaretAnimator::paint const):
* Source/WebCore/platform/CaretAnimator.h:
(WebCore::CaretAnimator::didStart):
(WebCore::currentTimeSinceEpoch): Deleted.
(WebCore::platformCaretColor): Deleted.
* Source/WebCore/platform/DictationCaretAnimator.cpp:
(WebCore::keyframe):
(WebCore::DictationCaretAnimator::keyframeCount const):
(WebCore::DictationCaretAnimator::keyframeTimeDelta const):
(WebCore::DictationCaretAnimator::updateAnimationProperties):
(WebCore::DictationCaretAnimator::start):
(WebCore::DictationCaretAnimator::paint const):
* Source/WebCore/platform/DictationCaretAnimator.h:
* Source/WebCore/platform/OpacityCaretAnimator.cpp:
(WebCore::OpacityCaretAnimator::updateAnimationProperties):
(WebCore::OpacityCaretAnimator::start):
(WebCore::OpacityCaretAnimator::paint const):
* Source/WebCore/platform/OpacityCaretAnimator.h:
* Source/WebCore/platform/SimpleCaretAnimator.cpp:
(WebCore::SimpleCaretAnimator::updateAnimationProperties):
(WebCore::SimpleCaretAnimator::start):
* Source/WebCore/platform/SimpleCaretAnimator.h:
* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::autocorrectionReplacementMarkerColor const):
* Source/WebKit/Platform/cocoa/_WKWebViewTextInputNotifications.mm:
(-[_WKWebViewTextInputNotifications dictationDidStart]):
(-[_WKWebViewTextInputNotifications dictationDidResume]):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::selectionDidChange):
(WebKit::WebViewImpl::updateCursorAccessoryPlacement):
(WebKit::WebViewImpl::updateCaretDecorationPlacement): Deleted.
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::getPlatformEditorState const):
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame caretColor]):

Canonical link: https://commits.webkit.org/265430@main
